assets/scss/components/_header.scss

```diff
@@ -6,6 +6,16 @@
     display: flex;
     flex-direction: column;
     background-color: $color-secondary-50;
-    position: relative;
-    z-index: $z-index-mid;
+    position: sticky;
+    top: 0;
+    transition: transform 0.3s ease-in-out;
+    z-index: $z-index-high;
+}
+
+.c-header.pushed {
+    transform: translateY(-100%);
+
+    @include bp(medium) {
+        transform: translateY(calc(-100% + 54px));
```

Comment:
I'd suggest to avoid using "magic numbers" and put it into a variable with a descriptive name